js/ui/windowManager.js

```diff
@@ -834,27 +834,44 @@ const WindowManager = new Lang.Class({
             return;
         }
 
-        let endX;
-        if (actor.x <= monitor.x) {
-            endX = monitor.x - actor.width;
-        } else {
-            endX = monitor.x + monitor.width;
-        }
+        let centered = (monitor.width - actor.width) * 0.5;
 
         actor.opacity = 255;
         actor.show();
 
-        Tweener.addTween(actor,
-                         { x: endX,
-                           time: WINDOW_ANIMATION_TIME,
-                           transition: "easeOutQuad",
-                           onComplete: onComplete,
-                           onCompleteScope: this,
-                           onCompleteParams: [shellwm, actor],
-                           onOverwrite: onOverwrite,
-                           onOverwriteScope: this,
-                           onOverwriteParams: [shellwm, actor]
-                         });
+        if (actor.x == centered) {
```

Comment:
Checking to see if the actor is centered on the monitor like this seems awfully fragile - can we check some other flag on the window?